src/main/java/org/isf/utils/jobjects/VoDateTextField.java

```diff
@@ -189,38 +197,46 @@ public void focusLost(FocusEvent e) throws IllegalArgumentException {
 			}
 		});
 	}
-	
+
 	/**
 	 * Returns a GregorianCalendar for date use
 	 */
 	public GregorianCalendar getDate() {
 		GregorianCalendar calendar = new GregorianCalendar();
 		calendar.setLenient(false);
-		if (type.equals("dd/mm/yyyy")) {
-			calendar.set(Integer.parseInt(getText().substring(6,10)),Integer.parseInt(getText().substring(3,5))-1,Integer.parseInt(getText().substring(0,2)));
-		} else if (type.equals("mm/dd/yyyy")) {
-			calendar.set(Integer.parseInt(getText().substring(6,10)),Integer.parseInt(getText().substring(0,2)) - 1,Integer.parseInt(getText().substring(3,5)));
-		} else if (type.equals("dd/mm/yy")) {
-			calendar.set(Integer.parseInt(getText().substring(6,8)) + 2000,Integer.parseInt(getText().substring(3,5))-1,Integer.parseInt(getText().substring(0,2)));
-		} else if (type.equals("mm/dd/yy")) {
-			calendar.set(Integer.parseInt(getText().substring(6,8)) + 2000,Integer.parseInt(getText().substring(0,2)) - 1,Integer.parseInt(getText().substring(3,5)));
+		if (type.equals(DATE_FORMAT_DD_MM_YYYY)) {
+			calendar.set(Integer.parseInt(getText().substring(6, 10)), Integer.parseInt(getText().substring(3, 5)) - 1,
+					Integer.parseInt(getText().substring(0, 2)));
+		} else if (type.equals(DATE_FORMAT_MM_DD_YYYY)) {
+			calendar.set(Integer.parseInt(getText().substring(6, 10)), Integer.parseInt(getText().substring(0, 2)) - 1,
+					Integer.parseInt(getText().substring(3, 5)));
+		} else if (type.equals(DATE_FORMAT_DD_MM_YY)) {
+			calendar.set(Integer.parseInt(getText().substring(6, 8)) + 2000, Integer.parseInt(getText().substring(3, 5)) - 1,
+					Integer.parseInt(getText().substring(0, 2)));
+		} else if (type.equals(DATE_FORMAT_MM_DD_YY)) {
+			calendar.set(Integer.parseInt(getText().substring(6, 8)) + 2000, Integer.parseInt(getText().substring(0, 2)) - 1,
+					Integer.parseInt(getText().substring(3, 5)));
 		}
 		return calendar;
 	}
 
+	public LocalDate getLocalDate() {
+		return Converters.convertToLocalDateTime(getDate()).toLocalDate();
+	}
+
 	public void setDate(GregorianCalendar time) {
 		String string;
 		if (time.get(Calendar.DAY_OF_MONTH) > 9) {
 			string = String.valueOf(time.get(Calendar.DAY_OF_MONTH));
 		} else {
-			string = "0" + String.valueOf(time.get(Calendar.DAY_OF_MONTH));
+			string = "0" + time.get(Calendar.DAY_OF_MONTH);
 		}
 		if (time.get(Calendar.MONTH) + 1 > 9) {
-			string += "/" + String.valueOf(time.get(Calendar.MONTH) + 1);
+			string += "/" + (time.get(Calendar.MONTH) + 1);
 		} else {
-			string += "/0" + String.valueOf(time.get(Calendar.MONTH) + 1);
+			string += "/0" + (time.get(Calendar.MONTH) + 1);
 		}
-		string += "/" + String.valueOf(time.get(Calendar.YEAR));
+		string += "/" + time.get(Calendar.YEAR);
 		currentDate = string;
 	}
```

Comment:
is this ever used now?

Reply:
yes in the LabBrowser module.  More analysis needs to be done to see if other related changes are needed.

Reply:
After the "big merge" I will go back and investigate all uses of `GregorianCalendar` and resolve these last little issues.

Reply:
Agreed! Thanks a lot!